### PR TITLE
<Box/> -  support space-separated spacing tokens

### PR DESCRIPTION
**Pull request by Zivilekin**
### 🔦 Summary

Resolved issue:

There was a bug that complex values + spacing tokens didn't work, for example:
margin="SP1 SP2"

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6281

**Review by EgoziE — APPROVED** [review]

File: src/Box/Box.js [hunk 1]
```diff
@@ -93,28 +106,32 @@ const Box = ({
...
+    classes.root,
+    {
+      inline,
 
-    // Alignment
-    [directions[direction]]: direction,
-    [horizontalAlignmentValues[align]]: align,
-    [verticalAlignmentValues[verticalAlign]]: verticalAlign,
-  });
+      // Alignment
```
Comment (mykas) [inline src/Box/Box.js:114]: Lets remove this comment here

File: src/Box/Box.js [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
-import styles from './Box.scss';
-
+import { st, classes } from './Box.st.css';
 import { filterObject } from '../utils/filterObject';
 import { spacingUnit, spacingTokens } from '../spacing';
 
 const directions = {
-  horizontal: styles.horizontal,
-  vertical: styles.vertical,
+  horizontal: 'horizontal',
```
Comment (mykas) [inline src/Box/Box.js:10]: We don't use these constants anymore, right?

File: src/Box/constants.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { spacingUnit } from '../spacing';
```
Comment (mykas) [inline src/Box/constants.js:1]: Lets import spacing units from `spacing.st.css` and delete this `spacing.js` file as it was used only by Box component.

File: src/Box/Box.js [hunk 4]
```diff
@@ -1,51 +1,47 @@
...
 
 /** In case the value is a number, it's multiplied by the defined spacing unit.
  *  Otherwise - there are three options:
  *   1. A Spacing Token - SP1, SP2, etc. - where the number is multiplied by the spacing unit.
  *   2. A predefined spacing value with semantic name (tiny, small, etc.)
  *   3. Space-separated values that are represented by a string (for example: "3px 3px")
  * */
-const formatSpacingValue = value => {
+const formatSingleSpacingValue = value => {
+  if (typeof value !== 'undefined') {
```
Comment (mykas) [inline src/Box/Box.js:22]:
```suggestion
  if (value !== undefined) {
```

File: src/Box/Box.js [hunk 5]
```diff
@@ -93,28 +89,30 @@ const Box = ({
...
+    padding: formatComplexSpacingValue(padding),
+    paddingTop: formatSingleSpacingValue(paddingTop),
+    paddingRight: formatSingleSpacingValue(paddingRight),
+    paddingBottom: formatSingleSpacingValue(paddingBottom),
+    paddingLeft: formatSingleSpacingValue(paddingLeft),
+    margin: formatComplexSpacingValue(margin),
+    marginTop: formatSingleSpacingValue(marginTop),
+    marginRight: formatSingleSpacingValue(marginRight),
+    marginBottom: formatSingleSpacingValue(marginBottom),
+    marginLeft: formatSingleSpacingValue(marginLeft),
```
Comment (shlomitc) [inline src/Box/Box.js:115]: since these calculations will happen on each render, maybe it's better to memoize them

Comment (mykas) [reply src/Box/Box.js:115]: good idea!

**Review by mykas — APPROVED** [review]

**Pull request merged**